lib/compress.js

```diff
@@ -2264,6 +2264,7 @@ merge(Compressor.prototype, {
         def(AST_Constant, return_true);
         def(AST_Lambda, function(scope){
             var self = this;
+            if (self.inlined) return false;
```

Comment:
Is `.inlined` cleared between passes?

Reply:
Indeed - for both `AST_Defun` & `AST_Function`
https://github.com/mishoo/UglifyJS2/blob/206a54a7461b76683d690fd7016943fcb461a4fa/lib/compress.js#L375-L394

Reply:
Cool. Thanks.

Reply:
Just so we are on the same page - `.inlined` is introduced to suppress further optimisations that are dangerous after a function is substituted into a different scope, since things like `.enclosed` won't be up-to-date.

So we wait for the next `passes` when a proper `.figure_out_scope()` is performed before resuming optimisations on the substituted function.

Reply:
Perfect - I assumed that was why you introduced it. The function state would be in flux mid-pass.